pkg/nsx/services/subnet/subnet.go

```diff
@@ -138,6 +143,11 @@ func (service *SubnetService) createOrUpdateSubnet(obj client.Object, nsxSubnet
 		Jitter:   0,
 		Steps:    6,
 	}
+
+	// For update case, wait for some time to make sure the realized state is updated.
+	if isUpdate {
+		time.Sleep(1 * time.Second)
+	}
```

Comment:
it's better that NSX fix that issue. After NSX patch API called, the realized status should be updated in sync mode instead

Reply:
Opened a bug for NSX team, let's see how they rely on this observations, thanks